src/controllers/queries/index.ts

```diff
@@ -184,6 +203,114 @@ export class QueriesController extends HTMLController {
     )
   }
 
+  /**
+   * Retrieves the query response page
+   */
+  @SuccessResponse(200)
+  @Get('/scope-3-carbon-consumption/{queryId}/response')
+  public async scope3CarbonConsumptionResponse(@Path() queryId: UUID): Promise<HTML> {
+    this.logger.debug('query response page requested %j', { queryId })
+    const [query] = await this.db.get('query', { id: queryId })
+
+    if (!query) {
+      throw new NotFoundError(`There has been an issue retrieving the query.`)
+    }
+
+    const [connection] = await this.db.get('connection', { id: query.connection_id })
+    if (!connection) {
+      throw new InvalidInputError(`There has been an issue retrieving the connection.`)
+    }
+
+    return this.html(
+      this.scope3CarbonConsumptionResponseTemplates.newScope3CarbonConsumptionResponseFormPage(
+        'form',
+        connection,
+        queryId,
+        query.details['quantity'],
+        query.details['productId']
+      )
+    )
+  }
+
+  /**
+   * Submits the query response page
+   */
+  @SuccessResponse(200)
+  @Post('/scope-3-carbon-consumption/{queryId}/response')
+  public async scope3CarbonConsumptionResponseSubmit(
+    @Path() queryId: UUID,
+    @Body()
+    body: {
+      companyId: UUID
+      action: 'success'
+      totalScope3CarbonEmissions: string
+      partialResponse?: number
+    }
+  ): Promise<HTML> {
+    this.logger.debug('query page requested')
+
+    const [connection] = await this.db.get('connection', { id: body.companyId, status: 'verified_both' }, [
+      ['updated_at', 'desc'],
+    ])
+    if (!connection) {
+      throw new NotFoundError(`Invalid connection ${body.companyId}`)
+    }
+    if (!connection.agent_connection_id || connection.status !== 'verified_both') {
+      throw new NotFoundError(`Cannot query unverified connection`)
+    }
+    const [queryRow] = await this.db.get('query', { id: queryId })
+    if (!queryRow.response_id) {
+      throw new NotFoundError(`Missing queryId to respond to.`)
+    }
```

Comment:
These should not be `NotFoundError`. `NotFoundError` is a `404` response which should be reserved for non existant path resources. In this case that's only for `queryId` values not found in the database. This also happens to be a case we're not catching.

In short we need 

* an extra check for the truthyness of `queryRow`. Is falsey this should throw `NotFoundError`.
* replace the existing throws with a `InvalidInputError`

Reply:
Should be updated, not sure if you will like it, but broken down into 
- queryRow = NotFound
- invalid inputs for others like "status", "response_id" and etc.